appcues/src/test/java/com/appcues/statemachine/effects/PresentationEffectTest.kt

```diff
@@ -0,0 +1,221 @@
+package com.appcues.statemachine.effects
+
+import com.appcues.action.ActionProcessor
+import com.appcues.action.ExperienceAction
+import com.appcues.data.model.Experience
+import com.appcues.data.model.ExperienceTrigger.Qualification
+import com.appcues.data.model.ExperienceTrigger.ShowCall
+import com.appcues.statemachine.Action.RenderStep
+import com.appcues.statemachine.Action.ReportError
+import com.appcues.statemachine.Error
+import com.appcues.trait.AppcuesTraitException
+import com.appcues.trait.MetadataSettingTrait
+import com.appcues.trait.PresentingTrait
+import com.google.common.truth.Truth.assertThat
+import io.mockk.Called
+import io.mockk.coVerify
+import io.mockk.coVerifySequence
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import io.mockk.verifySequence
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+internal class PresentationEffectTest {
```

Comment:
PresentationEffect is the complex effect, there is a bunch of tests that represent things that was previously being done by BeginningStepState